sn_node/src/node/bootstrap/join.rs

```diff
@@ -358,22 +358,23 @@ impl<'a> Joiner<'a> {
             return None;
         }
         let _ = self.retry_responses_cache.insert(sender, expected_age);
-        let mut responded = 0;
         let mut expected_ages: BTreeMap<u8, u8> = Default::default();
         for (elder, expected_age) in self.retry_responses_cache.iter() {
             if elders.contains(elder) {
-                responded += 1;
                 *expected_ages.entry(*expected_age).or_insert(0) += 1;
             }
         }
 
-        if responded > elders.len() / 3 && !self.aggregated {
-            let (voted_expected_age, _count) = expected_ages
+        // To avoid restarting too quick, the voted age must got supermajority votes.
+        // In case there is split votes, leave to the overall join_timeout to restart.
+        if !self.aggregated {
+            let threshold = 2 * elders.len() / 3;
+            let voted_expected_age = expected_ages
                 .iter()
-                // In case two ages got the same votes, we prefer the lower age
-                .max_by_key(|(age, count)| (*count, -(**age as i32)))?;
-            if self.node.age() != *voted_expected_age {
-                return Some(*voted_expected_age);
+                .find(|(_age, count)| **count >= threshold as u8)
+                .map(|(age, _count)| *age)?;
```

Comment:
Don't you want to prefer the smaller age if you have multiple ages that have passed the threshold? i.e.
```suggestion
                .filter(|(_age, count)| **count >= threshold as u8)
                // In case two ages got the same votes, we prefer the lower age
                .max_by_key(|(age, count)| (*count, -(**age as i32)))?;
```

Reply:
the threshold is 2/3 of the elders.len(), and the votes shall all from the same elders, which means there shall only be one `age` got more than 2/3 elders' votes.
so no need to have the `max_by_key` anymore

Reply:
we talked through this, the point is that we always track the most-recent votes from each elder, so it's impossible to get multiple super-majorities here.